### DISCUSSION TITLE
Trouble with transitions and highlight.js

### DISCUSSION THREAD

**Discussion by tcober** [nuxt/nuxt#18428] [Questions]
So I made a composable for highlighting code blocks on my blog pages using highlights.js. It worked great until I added transitions. Then, for some reason, it was firing before, I guess, the transition would allow it to see the code blocks. I fixed this by firing the composable function in a `onAfterEnter` hook in the `definePageMeta()` function on my blog pages. This is not a great solution for a couple of reasons.

1. It calls the highlights.js on every page regardless of whether it has a code block or not and more importantly
2. It doesn't work when someone directly visits a blog page because the transition doesn't run on the first page you enter the site on.

Not sure what to do. Anyone got any suggestions? Am I going about this all wrong?

**Comment by Baroshem** [comment]
Hey,

Not sure what could be the issue here. Could you create a reproduction example by using Stackblitz with Nuxt example?

It may be that it just needs these transitions to be run on another lifecycle hook(onBeforeMount) or maybe it is something more advanced so reproduction would definitely help :)

**Reply by tcober** [reply]
Have done a bunch of research and experimenting around this and I appear to be running into this bug: https://github.com/nuxt/nuxt/issues/13471

and my workaround doesn't work because of this bug:
https://github.com/nuxt/nuxt/issues/14715

which is an upstream bug with Vue 3 that is logged here:
https://github.com/vuejs/core/issues/6951

So I'm going to just try and roll my own transition with some css and lifcycle hooks and am hoping I can get something to work that way and maybe the bugs will get fixed at some point and I can do the documented way.
